module/zfs/vdev_raidz.c

```diff
@@ -1970,8 +1970,15 @@ vdev_raidz_open(vdev_t *vd, uint64_t *asize, uint64_t *max_asize,
 		    cvd->vdev_physical_ashift);
 	}
 
-	*asize *= vd->vdev_children;
-	*max_asize *= vd->vdev_children;
+	if (vd->vdev_rz_expanding) {
+		*asize *= vd->vdev_children - 1;
+		*max_asize *= vd->vdev_children - 1;
+
+		vd->vdev_min_asize = *asize;
```

Comment:
Why do we need to set `vdev_min_asize` explicitly here?  I think it should be done when vdev_open() calls vdev_set_min_asize(), which happens after vdev_raidz_open().  Is there a reason that it needs to be set before that?